sdk/evaluation/azure-ai-evaluation/azure/ai/evaluation/_converters/_ai_services.py

```diff
@@ -0,0 +1,468 @@
+import json
+from concurrent.futures import ThreadPoolExecutor, as_completed
+
+from azure.ai.projects import AIProjectClient
+from azure.ai.projects.models import (
+    ThreadRun,
+    RunStep,
+    RunStepToolCallDetails,
+    FunctionDefinition,
+    ListSortOrder,
+)
+
+from typing import List
+
+# Constants.
+from ._models import _USER, _AGENT, _TOOL, _TOOL_CALL, _TOOL_CALLS, _FUNCTION
+
+# Message instances.
+from ._models import Message, SystemMessage, UserMessage, AssistantMessage, ToolCall
+
+# Intermediate definitions to hold results.
+from ._models import ToolDefinition, ConvertedResult
+
+# Utilities.
+from ._models import break_tool_call_into_messages, convert_message
+
+
+class AIAgentConverter:
+    """
+    A converter for AI agent data.
+
+    :param project_client: The AI project client used for API interactions.
+    :type project_client: AIProjectClient
+    """
+
+    def __init__(self, project_client: AIProjectClient):
+        """
+        Initializes the AIAgentConverter with the given AI project client.
+
+        :param project_client: The AI project client used for API interactions.
+        :type project_client: AIProjectClient
+        """
+        self.project_client = project_client
+
+    def __list_messages_chronological(self, thread_id: str):
+        """
+        Lists messages in chronological order for a given thread.
+
+        :param thread_id: The ID of the thread.
+        :type thread_id: str
+        :return: A list of messages in chronological order.
+        :rtype: List[Message]
+        """
+        messages = self.project_client.agents.list_messages(
+            thread_id=thread_id, limit=100, order=ListSortOrder.ASCENDING
```

Comment:
Is this limit temporary ?

Reply:
Good catch, no, 100 is the maximum AI services API can return at once. I've added pagination for both the messages and run step details, which will go through and get results as long as we get `has_more` with a new anchor.